packages/block-library/src/paragraph/save.js

```diff
@@ -7,11 +7,15 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { RichText, useBlockProps } from '@wordpress/block-editor';
+import { isRTL } from '@wordpress/i18n';
 
 export default function save( { attributes } ) {
 	const { align, content, dropCap, direction } = attributes;
 	const className = classnames( {
-		'has-drop-cap': dropCap,
+		'has-drop-cap':
+			align === ( isRTL() ? 'left' : 'right' ) || align === 'center'
+				? false
+				: dropCap,
```

Comment:
This doesn't work well, there are block validation error when you switch between ltr and rtl. I'll try something else.